10up/ElasticPress
# Fix remove from index on update

This change fixed an issue that I came across: Setting a password on a post and updating it from the admin page did _not_ remove it from the index, unless I hit the update button a second time.

Details about the bug can be found here: #3936

This PR makes the last running filter have the final say about adding the post to the index or not.

I found this issue on WordPress 6.4.3 using ElasticPress 4.7.2. As far as I can see the relevant code has not changed since. I did not test this change on the latest ElasticPress version yet.

I tried to whip up a Cypress test to demonstrate the issue but it does not seem to work correctly. I am not familiar with Cypress, some help is appreciated.

Closes #3936

### How to test the Change
1. Create and publish a post
2. Check that the post is in de index
3. Change visibility, set a password
4. Update the post
5. Check that the post is no longer in the index

### Changelog Entry

> Fixed - Failure to remove post from index during update for some WordPress versions

### Credits
Props @dtakken


### Checklist:
